design.md

```diff
@@ -2,12 +2,27 @@
 
 ## Who is the target for this design, e.g., are you assuming any knowledge on the part of the language users?
 
+This design is for CS5 students. We assume that students have a basic understanding of scope and syntax as well as reading the picobot assignment. At the very least, someone should be able to read a program and understand what steps picobot should take.
```

Comment:
Scope is actually a pretty complicated subject and assuming that CS5 students have a basic understanding of it in their first week is probably dangerous. On the other hand, it doesn't look like your syntax utilizes scope, just the idea of blocks which should be good